MapboxCoreNavigation/FreeDriveLocationManager.swift

```diff
@@ -0,0 +1,134 @@
+import MapboxNavigationNative
+import CoreLocation
+import MapboxDirections
+import MapboxAccounts
+
+public protocol FreeDriveDebugInfoListener: AnyObject {
+    func didGet(location: CLLocation, with matches: [MapMatch], for rawLocation: CLLocation)
+}
+
+open class FreeDriveLocationManager: NavigationLocationManager, CLLocationManagerDelegate {
+    public weak var debugInfoListener: FreeDriveDebugInfoListener? {
+        get {
+            proxyDelegate?.debugDelegate
+        }
+        set {
+            proxyDelegate?.debugDelegate = newValue
+        }
+    }
+
+    private var proxyDelegate: ProxyDelegate?
+
+    public required init(navigator: Navigator? = nil) {
+        proxyDelegate = ProxyDelegate()
+        super.init()
+
+        if let navigator = navigator {
+            proxyDelegate?.navNative = navigator
+        } else {
+            let tilesVersion = RouteTilesVersion(with: Directions.shared.credentials)
+            tilesVersion.getAvailableVersions { availableVersions in
+                if let latestVersion = availableVersions.last {
+                    tilesVersion.currentVersion = latestVersion
+                    let navigator = self.createNavigator(withTilesVersion: latestVersion)
+                    self.proxyDelegate?.navNative = navigator
+                }
+            }
+        }
+
+        super.delegate = proxyDelegate
+    }
+
+    private func createNavigator(withTilesVersion tilesVersion: String) -> Navigator {
+        let settingsProfile = SettingsProfile(application: ProfileApplication.kMobile, platform: ProfilePlatform.KIOS)
+        let navigator = Navigator(profile: settingsProfile, config: NavigatorConfig() , customConfig: "")
+        let host = Directions.shared.credentials.host.absoluteString
+        let publicToken = Directions.shared.credentials.accessToken ?? ""
+        assert(publicToken != "")
+        let endpointConfig = TileEndpointConfiguration(
+            host: host,
+            version: tilesVersion,
+            token: publicToken,
+            userAgent: "",
+            navigatorVersion: "", skuTokenSource: SkuTokenProvider(with: DirectionsCredentials())
```

Comment:
Per https://github.com/mapbox/mapbox-navigation-android/pull/2326#issuecomment-567844536, it may actually be desirable to pass in an empty string, rather than the Core Navigator bundle version, to avoid side effects. (We really don’t want the navigator to try to use cURL, which doesn’t exist on iOS.)